packages/blade/src/utils/styled-props/useStyledProps.web.tsx

```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import type { CSSObject } from 'styled-components';
+import { getStyledProps } from './getStyledProps';
+import { getBaseBoxStyles, getDependencyProps } from '~components/Box/BaseBox/getBaseBoxStyles';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const useStyledProps = (props: Record<string, any>): CSSObject => {
+  const styledPropsStyles = getStyledProps(props);
+  const styledPropsMemoDependency = getDependencyProps(styledPropsStyles);
+  const styledPropsCSSObject = React.useMemo(
+    () => getBaseBoxStyles({ ...styledPropsStyles, theme: props.theme }),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [styledPropsMemoDependency],
```

Comment:
oh currently the theme isn't. In what scenario, theme would change 🤔?

Reply:
Created this PR to handle theme changes - https://github.com/razorpay/blade/pull/1031

Will merge it at the end after layouts since we're right now only using theme switching in our recipe.